handlers/chikka/chikka.go

```diff
@@ -1,3 +1,148 @@
 package chikka
 
-/* no logs */
+import (
+	"context"
+	"fmt"
+	"github.com/nyaruka/courier/utils"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/gocommon/urns"
+)
+
+var (
+	sendURL      = "https://post.chikka.com/smsapi/request"
+	maxMsgLength = 160
+)
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+func newHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("CK"), "Chikka")}
+}
+
+func init() {
+	courier.RegisterHandler(newHandler())
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	return s.AddHandlerRoute(h, http.MethodPost, "receive", h.ReceiveMessage)
+}
+
+type moMsg struct {
+	MessageType  string  `name:"message_type" validate:"required"`
+	RequestID    string  `name:"request_id"`
+	MobileNumber string  `name:"mobile_number"`
+	Message      string  `name:"message"`
+	Timestamp    float64 `name:"timestamp"`
+	MessageID    int64   `name:"message_id"`
+	Status       string  `name:"status"`
+}
+
+var statusMapping = map[string]courier.MsgStatusValue{
+	"SENT":   courier.MsgSent,
+	"FAILED": courier.MsgFailed,
+}
+
+// ReceiveMessage is our HTTP handler function for incoming messages
+func (h *handler) ReceiveMessage(ctx context.Context, channel courier.Channel, w http.ResponseWriter, r *http.Request) ([]courier.Event, error) {
+	ckRequest := &moMsg{}
+	err := handlers.DecodeAndValidateForm(ckRequest, r)
+	if err != nil {
+		return nil, err
+	}
+
+	if ckRequest.MessageType == "outgoing" {
+
+		msgStatus, found := statusMapping[ckRequest.Status]
+		if !found {
+			return nil, courier.WriteAndLogRequestError(ctx, w, r, channel, fmt.Errorf(`unknown status '%s', must be either 'SENT' or 'FAILED'`, ckRequest.Status))
+		}
+
+		// write our status
+		status := h.Backend().NewMsgStatusForID(channel, courier.NewMsgID(ckRequest.MessageID), msgStatus)
+		err = h.Backend().WriteMsgStatus(ctx, status)
+		if err != nil {
```

Comment:
Should also deal with the case where we get an error about the msg not existing.